### PR TITLE
MLT-0075 Order Domain changes

### PR DESCRIPTION
**Pull request by anotheroneofthese**
Replaces Receiver with two new fields in the Order model

### CODE REVIEW

File: src/main/kotlin/no/nb/mlt/wls/domain/WLSService.kt [hunk 1]
```diff
@@ -163,7 +164,7 @@ class WLSService(
 
         val order = getOrderOrThrow(hostName, hostOrderId)
 
-        val updatedOrder = order.updateOrder(itemHostIds, callbackUrl, orderType, receiver)
+        val updatedOrder = order.updateOrder(itemHostIds, callbackUrl, orderType, contactPerson)
```
Comment (MormonJesus69420) [inline src/main/kotlin/no/nb/mlt/wls/domain/WLSService.kt:167]: Should be able to edit address too, no?

File: src/main/kotlin/no/nb/mlt/wls/domain/model/Order.kt [hunk 2]
```diff
@@ -98,14 +99,14 @@ data class Order(
         itemIds: List<String>,
         callbackUrl: String,
         orderType: Type,
-        receiver: Receiver
+        contactPerson: String
```
Comment (MormonJesus69420) [inline src/main/kotlin/no/nb/mlt/wls/domain/model/Order.kt:102]: Same comment here, address should be editable

File: src/main/kotlin/no/nb/mlt/wls/domain/model/Order.kt [hunk 3]
```diff
@@ -153,10 +154,37 @@ data class Order(
...
+        val name: String?,
+        val addressLine1: String?,
+        val addressLine2: String?,
+        val zipcode: String?,
+        val city: String?,
+        val state: String?,
+        val country: String?
+    ) {
+        fun validate() {
+            if (name?.isBlank() == true) {
```
Comment (MormonJesus69420) [inline src/main/kotlin/no/nb/mlt/wls/domain/model/Order.kt:167]: == true is redundant

File: src/main/kotlin/no/nb/mlt/wls/infrastructure/synq/SynqOrderPayload.kt [hunk 4]
```diff
@@ -40,6 +40,7 @@ data class SynqOrderPayload(
     }
 }
 
+// TODO - Should this be expanded with more fields?
```
Comment (MormonJesus69420) [inline src/main/kotlin/no/nb/mlt/wls/infrastructure/synq/SynqOrderPayload.kt:43]: Yes

**Review by MormonJesus69420 — CHANGES_REQUESTED** [review]

File: src/main/kotlin/no/nb/mlt/wls/domain/model/Order.kt (hunk 3 above)
Comment (anotheroneofthese) [reply src/main/kotlin/no/nb/mlt/wls/domain/model/Order.kt:167]: Nullability is weird

File: src/main/kotlin/no/nb/mlt/wls/infrastructure/callbacks/NotificationOrderPayload.kt [hunk 5]
```diff
@@ -59,9 +65,13 @@ data class NotificationOrderPayload(
...
     @Schema(
-        description = "Who's the receiver of the material in the order."
+        description = "The person to contact regarding matters about the order"
     )
-    val receiver: Order.Receiver,
+    val contactPerson: String,
+    @Schema(
+        description = "The address of the receiver of the material in the order."
+    )
+    val address: Order.Address?,
```
Comment (MormonJesus69420) [inline src/main/kotlin/no/nb/mlt/wls/infrastructure/callbacks/NotificationOrderPayload.kt:74]: Notice: note not noted in notification, not nice, no, no... Note the note in notification now.

File: src/main/kotlin/no/nb/mlt/wls/infrastructure/synq/SynqOrderPayload.kt [hunk 6]
```diff
@@ -43,10 +43,16 @@ data class SynqOrderPayload(
 data class ShippingAddress(
     val address: Address
 ) {
+    // TODO - Should all the fields here be annotated?
```
Comment (MormonJesus69420) [inline src/main/kotlin/no/nb/mlt/wls/infrastructure/synq/SynqOrderPayload.kt:46]: Yeah, I think the address as a whole and its fields should not be included if null

**Review by MormonJesus69420 — CHANGES_REQUESTED** [review]

File: src/main/kotlin/no/nb/mlt/wls/infrastructure/synq/SynqOrderPayload.kt (hunk 6 above)
Comment (anotheroneofthese) [reply src/main/kotlin/no/nb/mlt/wls/infrastructure/synq/SynqOrderPayload.kt]: The ShippingAddress is currently not nullable, due to `contactPerson` being a required field for the SynQ payload. Should this information be sent to storage? If not, then we can make it an optional field

**Comment by MormonJesus69420** [comment]
I dunno if you and Tom have discussed it, but shouldn't the  person in the address be nullable as well?

**Review by MormonJesus69420 — APPROVED** [review]
Looks good, why are ports in compose file written in quotes now?

EDIT: oh okay so apparently anything in YAML written as xx:xx will be parsed as a base 60 number if values are under 60, weird but okay

**Comment by anotheroneofthese** [comment]
New best practice, according to https://stackoverflow.com/questions/58810789/quotes-on-docker-compose-yml-ports-make-any-difference
image
Docker doesn't seem to reflect this in their specification anymore. They might have changed it, although there seems to be questions floating around where the solution usually is stringifying the ports

**Comment by MormonJesus69420** [comment]
image
https://yaml.org/type/float.html

Yeah it is still recommended practice to write HOST:CONTAINER as strings

**Pull request merged**